discojs/src/aggregator.spec.ts

```diff
@@ -35,10 +78,36 @@ AGGREGATORS.forEach(([name, Aggregator]) =>
           aggregator.add(`client ${i}`, WeightsContainer.of([i]), 0, r);
 
       await results; // nothing to test
-
+    
       expect(aggregator.round).to.equal(1);
     });
 
+    it("does not move forward with not enough contributions", async () => {
+      const aggregator = new Aggregator();
+      aggregator.setNodes(Set.of("client 0", "client 1", "client 2"));
+
+      const results = aggregator.receiveResult();
+
+      for (let i = 0; i < 2; i++)
+        for (let r = 0; r < aggregator.communicationRounds; r++)
+          aggregator.add(`client ${i}`, WeightsContainer.of([i]), 0, r);
+
+      await results; 
+
+      expect(aggregator.round).to.equal(0);
+    });
+
+    it("Adding at the wrong round does not count", () => {
```

Comment:
reads more nicely. also, you can keep the test itself smaller by not having three clients but only one. less code == faster to understand => less bug.

```suggestion
    it("drops contributions for futur rounds", () => {
```